src/pdmat.jl

```diff
@@ -1,19 +1,19 @@
 """
 Full positive definite matrix together with a Cholesky factorization object.
 """
-struct PDMat{T<:Real,S<:AbstractMatrix} <: AbstractPDMat{T}
+struct PDMat{T<:Real,S<:AbstractMatrix,C<:Factorization} <: AbstractPDMat{T}
     mat::S
-    chol::Cholesky{T,S}
+    chol::C
 
-    PDMat{T,S}(m::AbstractMatrix{T},c::Cholesky{T,S}) where {T,S} = new{T,S}(m,c)
+    PDMat{T,S,C}(m::AbstractMatrix{T}, c::Factorization) where {T,S,C} = new{T,S,C}(m,c)
```

Comment:
Another case where the previous implementation was not ideal and we might as well fix it now: the `checksquare` and `size` checks on lines 12-15 should probably move here, into the inner constructor. Note that with the current implementation, it's possible to do this:
```julia
julia> chol = Cholesky(LowerTriangular([2 0; 1 2]))
Cholesky{Int64, Matrix{Int64}}
L factor:
2×2 LowerTriangular{Int64, Matrix{Int64}}:
 2  ⋅
 1  2

julia> PDMat{Int,Matrix{Int}}(rand(Int, 7, 2), chol)
7×7 PDMat{Int64, Matrix{Int64}}:
  6099129454893146728  -6401227969804411468  #undef  #undef  #undef  #undef  #undef
   320386356076645564   5388511647264862786  #undef  #undef  #undef  #undef  #undef
 -1586394843294801620   -539204836094727972  #undef  #undef  #undef  #undef  #undef
  5363274486172883276  -1644246604771485866  #undef  #undef  #undef  #undef  #undef
 -7714840428325231882   5088781302322332903  #undef  #undef  #undef  #undef  #undef
   371206992283242625    280269229271602249  #undef  #undef  #undef  #undef  #undef
 -4201226691849289285  -7373499862429006397  #undef  #undef  #undef  #undef  #undef
```
which is rather broken. It's much safer to put sanity checks into the inner constructor.

Reply:
The current implementation was intended to provide a way to circumvent this check (since e.g. Zygote doesn't like them: https://github.com/JuliaStats/Distributions.jl/pull/1492) but in retrospect maybe this was not a good idea.

Reply:
I moved the checks to the inner constructor.